devito/data/decomposition.py

```diff
@@ -239,30 +255,32 @@ def index_glb_to_loc(self, *args, rel=True):
                 elif isinstance(glb_idx, slice):
                     if self.loc_empty:
                         return slice(-1, -3)
-                    if glb_idx.step >= 0 and glb_idx.stop == self.glb_min:
-                        glb_idx_min = self.glb_min if glb_idx.start is None \
+                    if glb_idx.step >= 0 and glb_idx.stop == glb_min:
+                        glb_idx_min = glb_min if glb_idx.start is None \
                             else glb_idx.start
-                        glb_idx_max = self.glb_min
+                        glb_idx_max = glb_min
                         retfunc = lambda a, b: slice(a, b, glb_idx.step)
                     elif glb_idx.step >= 0:
-                        glb_idx_min = self.glb_min if glb_idx.start is None \
+                        glb_idx_min = glb_min if glb_idx.start is None \
                             else glb_idx.start
-                        glb_idx_max = self.glb_max if glb_idx.stop is None \
+                        glb_idx_max = glb_max \
+                            if glb_idx.stop is None \
                             else glb_idx.stop-1
                         retfunc = lambda a, b: slice(a, b + 1, glb_idx.step)
                     else:
-                        glb_idx_min = self.glb_min if glb_idx.stop is None \
+                        glb_idx_min = glb_min if glb_idx.stop is None \
                             else glb_idx.stop+1
-                        glb_idx_max = self.glb_max if glb_idx.start is None \
+                        glb_idx_max = glb_max if glb_idx.start is None \
                             else glb_idx.start
                         retfunc = lambda a, b: slice(b, a - 1, glb_idx.step)
```

Comment:
In its current state, this is unreadable. I will return with a hopefully more meaningful suggestion....